test/e2e/shared/suite.go

```diff
@@ -75,17 +75,24 @@ func Node1BeforeSuite(e2eCtx *E2EContext) []byte {
 		sourceTemplate, err := os.ReadFile(f)
 		Expect(err).NotTo(HaveOccurred())
 
-		platformKustomization, err := os.ReadFile(filepath.Join(e2eCtx.Settings.DataFolder, "ci-artifacts-platform-kustomization.yaml"))
-		Expect(err).NotTo(HaveOccurred())
-
-		ciTemplate, err := kubernetesversions.GenerateCIArtifactsInjectedTemplateForDebian(
-			kubernetesversions.GenerateCIArtifactsInjectedTemplateForDebianInput{
-				ArtifactsDirectory:    e2eCtx.Settings.ArtifactFolder,
-				SourceTemplate:        sourceTemplate,
-				PlatformKustomization: platformKustomization,
-			},
-		)
-		Expect(err).NotTo(HaveOccurred())
+		ciTemplate := f
+
+		// For now, we skip the Debian artifacts template injection for Flatcar templates as they rely on 'apt' command
+		// and want to create files under /usr/local/bin/ - which can be an issue for immutable OS where /usr is
+		// read-only.
+		if !strings.Contains(f, "flatcar") {
```

Comment:
Instead of this, I suggest putting templates that should not have the "ci-artifacts" injected in a separate folder. Then we can do the `GenerateCIArtifactsInjectedTemplateForDebian` just like before since it only checks one folder. And for templates that don't need this injection we just copy them straight to `_artifacts` (where the "ci-artifacts" injected templates are written).